src/ot1d.jl

```diff
@@ -0,0 +1,283 @@
+"""
+otCost1d(c,μ::Distributions.UnivariateDistribution,
+           ν::Distributions.UnivariateDistribution)
+
+Calculates the Optimal Transport Cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where `h` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+"""
+function otCost1d(c,μ::Distributions.UnivariateDistribution,ν::Distributions.UnivariateDistribution)
+    g(μ,ν,x) = c(quantile(μ,x),quantile(ν,x))
+    f(x) = g(μ,ν,x)
+    return quadgk(f,0,1)[1]
+end
+
+"""
+otPlan1d(c,μ::Distributions.UnivariateDistribution,
+            ν::Distributions.UnivariateDistribution)
+
+Calculates the Optimal Transport Plan between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(|x-y|)``
+where ``h`` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+
+Returns the Optimal Transport Plan as a function
+  ```math
+  T(x)=F_\\nu^{-1}(F_\\mu(x)).
+  ```
```

Comment:
Style:
````suggestion
The optimal transport map ``T`` is returned as a function. It is defined as
```math
T(x) = F_{\\nu}^{-1}(F_{\\mu}(x)),
```
where ``F_{\\mu}`` and ``F_{\\nu}`` are the cumulative distribution functions of
distributions ``\\mu`` and ``\\nu``, respectively.
````
Maybe one should also rename `T` here to `\gamma` to make the notion consistent throughout the package:

````suggestion
The optimal transport map ``\\gamma`` is returned as a function. It is defined as
```math
\\gamma(x) = F_{\\nu}^{-1}(F_{\\mu}(x)),
```
where ``F_{\\mu}`` and ``F_{\\nu}`` are the cumulative distribution functions of
distributions ``\\mu`` and ``\\nu``, respectively.
````